From: Hannes Reinecke <hare@suse.de>
Subject: INQUIRY result too short (5) message flood
References: bnc#432535
Patch-mainline: not yet

During installation with vioserver/vioclient lots of
scsi scan: INQUIRY result too short (5), using 36
messages are printed. These really should be ratelimited.

Signed-off-by: Hannes Reinecke <hare@suse.de>

---
 drivers/scsi/scsi_scan.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/scsi/scsi_scan.c
+++ b/drivers/scsi/scsi_scan.c
@@ -696,7 +696,7 @@ static int scsi_probe_lun(struct scsi_de
 	 * and displaying garbage for the Vendor, Product, or Revision
 	 * strings.
 	 */
-	if (sdev->inquiry_len < 36) {
+	if (sdev->inquiry_len < 36 && printk_ratelimit()) {
 		sdev_printk(KERN_INFO, sdev,
			    "scsi scan: INQUIRY result too short (%d),"
			    " using 36\n", sdev->inquiry_len);
